From 613a4238f3a091b0cf2fe91110ed80851645f04e Mon Sep 17 00:00:00 2001
From: gavdov38 <gavin.a.docherty@gmail.com>
Date: Tue, 22 Sep 2026 08:06:20 +0100
Subject: [PATCH] selinux_hide.c: hide adb_root, su and fsck_untrusted
 sys_admin detections credit to backslashxx as this just implements his
 small_module in the kernel

---
 kernel/feature/selinux_hide.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/kernel/feature/selinux_hide.c b/kernel/feature/selinux_hide.c
index d342d42aa6ae..afcc30938872 100644
--- a/kernel/feature/selinux_hide.c
+++ b/kernel/feature/selinux_hide.c
@@ -299,9 +299,14 @@ init_hooks:; // apply_kernelsu_rules_fn
 	const char *init_adb_args[] = { "init", "adb_data_file", NULL };
 	ksu_add_shit_to_list(KSU_SEPOLICY_CMD_NORMAL_PERM, init_adb_args);
 
-	// we move this to a module instead
-	// const char *adbroot_args[] = { "adbroot", NULL };
-	// ksu_add_shit_to_list(KSU_SEPOLICY_CMD_TYPE, adbroot_args);
+	const char *adbroot_args[] = { "adbroot", "domain" };
+	ksu_add_shit_to_list(KSU_SEPOLICY_CMD_TYPE, adbroot_args);
+
+	const char *fsck_sys_admin_args[] = { "fsck_untrusted", "fsck_untrusted", "capability", "sys_admin" };
+	ksu_add_shit_to_list(KSU_SEPOLICY_CMD_XPERM, fsck_sys_admin_args);
+
+	const char *shell_su_args[] = { "shell", "su", "process", "transition" };
+	ksu_add_shit_to_list(KSU_SEPOLICY_CMD_TYPE_TRANSITION, shell_su_args);
 
 	int tries = 0;
 try_again:
